enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,133 @@
+---
+title: improved-platform-tests
+authors:
+  - "@soltysh"
+reviewers:
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+api-approvers:
+  - None
+creation-date: 2022-11-16
+last-updated: 2022-11-28
+tracking-link:
+  - None
+replaces:
+  - "enhancements/testing/extended-platform-tests.md"
+---
+
+# Improved Platform Tests
+
+## Summary
+
+Separate OpenShift tests from the ones coming from kubernetes. By doing so we will:
+- shorten the time when we start running most up-to-date tests from kubernetes;
+- run all tests coming from kubernetes, especially those introduced in patch releases;
+- remove the necessity to vendor kubernetes tests directly;
+- simplify the [kubernetes update process](https://github.com/openshift/kubernetes/blob/master/REBASE.openshift.md).
+
+## Motivation
+
+Over the past three releases the time to deliver the [updated kubernetes (o/k)](https://github.com/openshift/kubernetes/)
+to our [tests (origin)](https://github.com/openshift/origin/) has increased significantly.
+The delta for these was:
+- OpenShift 4.10 / kubernetes 1.23 - 7 days (compare [o/k#1087](https://github.com/openshift/kubernetes/pull/1087)
+  vs [origin#26753](https://github.com/openshift/origin/pull/26753)).
+- OpenShift 4.11 / kubernetes 1.24 - 21 days (compare [o/k#1252](https://github.com/openshift/kubernetes/pull/1252)
+  vs [origin#27244](https://github.com/openshift/origin/pull/27244)).
+- OpenShift 4.12 / kubernetes 1.25 - 40 days (compare [o/k#1360](https://github.com/openshift/kubernetes/pull/1360)
+  vs [origin#27526](https://github.com/openshift/origin/pull/27526)).
+
+Moreover, in the past two releases the changes to origin repository were reverted
+at least once, due to multiple issues found, which also affected the increased
+delays.
+
+### User Stories
+
+- As an OpenShift developer, I want to be able to seamlessly run OpenShift
+  and kubernetes-specific tests using `openshift-tests run`.
+- As an OpenShift developer, I want to be able to run newest kubernetes tests
+  without the necessity to bump vendored code.
+- As an OpenShift developer, I want to be able to extend and monitor cluster
+  during tests execution.
+- As an OpenShift user, I want to be able to verify that my cluster is providing
+  the desired functionality running one of the provided test suites.
+
+### Goals
+
+1. Reduce the risk updating kubernetes caused by excessive co-located code.
+2. Minimize time when we start running newest kubernetes tests.
+3. Maintain `openshift-tests` API for running tests.
+
+### Non-Goals
+
+1. Abandon existing tests.
+2. Change the available functionality of `openshift-tests`.
```

Comment:
IIUC it is not a goal to change the current content of `openshift-tests` suites, mainly the conformance ones (`openshift/conformance` and `kubernetes/conformance`), is that correct?

My concern is the e2e tests from kubernetes/K8s suite are in `openshift/conformance`, this suite is a base suite used to validate external clusters by partners on VCSP program - and we understand that the e2e tests in this suite cover all needed to validate an OCP installation.
```
$ ./openshift-tests run openshift/conformance --dry-run |grep -F "$(./openshift-tests run kubernetes/conformance --dry-run |tail -n1)"
"[sig-storage] Subpath Atomic writer volumes should support subpaths with secret pod [Excluded:WindowsDocker] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]"
$ ./openshift-tests run openshift/conformance --dry-run |grep -F "$(./openshift-tests run kubernetes/conformance --dry-run |head -n1)"
"[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing mutating webhooks should work [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]"
$ ./openshift-tests run openshift/conformance --dry-run |grep -F "$(./openshift-tests run kubernetes/conformance --dry-run |shuf |head -n1)"
"[sig-api-machinery] Watchers should observe an object deletion if it stops meeting the requirements of the selector [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]"

```

Reply:
everything you said is true, i'm not clear what the concern is though?

we can meet the goal of having k8s tests run as part of the openshift/conformance suite w/o including the k8s tests in the openshift-tests binary.

Reply:
> we can meet the goal of having k8s tests run as part of the openshift/conformance suite w/o including the k8s tests in the openshift-tests binary.

That's what I would like to make sure I got correctly to track changes on the VCSP program/OPCT.

Overall the two steps in this proposal will bring a huge benefit.

Reply:
@mtulio correct, we will ensure the current workloads continue working, at least wrt k8s conformance tests.